lib/sheets.ts

```diff
@@ -0,0 +1,175 @@
+import * as fs from "fs";
+import { env } from "./core-exec";
+
+import { google } from 'googleapis';
+
+import { log } from './console';
+
+import { _await } from './core-exec';
+
+
+let token: any;
+let credentials: any;
+
+let oAuth2Client: any;
+let sheets: any;
+
+// parse token.js and credentials.js
+export function initialize() {
+    const tokenPath = env.GOOGLE_TOKEN_PATH;
+    const credPath = env.GOOGLE_CREDENTIAL_PATH;
+
+    const tokenFile = fs.readFileSync(tokenPath, 'utf8');
+    const credFile = fs.readFileSync(credPath, 'utf8');
+
+    token = JSON.parse(tokenFile);
+    credentials = JSON.parse(credFile);
+
+    const { client_secret, client_id, redirect_uris } = credentials.installed;
+    oAuth2Client = new google.auth.OAuth2(client_id, client_secret, redirect_uris[0]);
+    oAuth2Client.setCredentials(token);
+
+    sheets = google.sheets({ version: 'v4', oAuth2Client });
+
+    return;
+}
+
+/*
+
+function getNewToken(oAuth2Client) {
+
+    const SCOPES = ['https://www.googleapis.com/auth/spreadsheets'];
+
+    const authUrl = oAuth2Client.generateAuthUrl({
+        access_type: 'offline',
+        scope: SCOPES,
+    });
+
+}
+*/
+
+// Exported Class Spreadsheet
+//   Methods to append row
+//   Clear Sheet
+//   Read Row
+//   Read Cell
+
+export class Spreadsheet {
+
+    private spreadsheetID: string;
+    private name: string;
+
+    constructor(spreadsheetID: string, name: string) {
+        this.spreadsheetID = spreadsheetID;
+        this.name = name;
+    }
+
+    public clear() {
+        return _await(new Promise((resolve, reject) => {
+            sheets.spreadsheets.values.clear({
+                range: this.name,
+                spreadsheetId: this.spreadsheetID,
+            }, (err: any, result: any) => {
+                if (err) {
+                    // Handle error.
+                    reject(err);
+                } else {
+                    resolve(`${result.data} cells cleared.`);
```

Comment:
See the comment below about how this causes the function to return a string. This method should be `void`.